### PR TITLE
dashboard: Fix spelling

### PR DESCRIPTION
**Pull request by jsoref**
This is based on https://github.com/ceph/ceph/pull/46824 and https://github.com/ceph/ceph/pull/46824

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket -- I'm not sure.
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

Note: I'm more than happy to squash, but I'd like to get buy-in for the changes first. Generally for changes w/ >10 things there's at least one that someone finds objectionable. And it's much easier for me to drop / fix them before I squash than after (and similarly, if I need to rebase, which I did for the parent PR, it's considerably more error prone if it's already squashed).

I'll annotate this PR shortly w/ the things that I think are most likely to be dropped/changed...

### CODE REVIEW

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet
Comment (jsoref) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:675]: I seem to recall a discussion about not wanting to make certain kinds of changes to the prometheus content. Sadly GitHub and my browser are not cooperating atm, so I can't find that in the meta PR.

Offhand, this is a relatively safe change since it's just human facing

File: monitoring/ceph-mixin/tests_alerts/test_alerts.yml
Comment (jsoref) [inline monitoring/ceph-mixin/tests_alerts/test_alerts.yml:1757]: The alert names are more likely to be problematic, so I think it's likely the project will want to drop this

File: monitoring/ceph-mixin/tests_dashboards/util.py
Comment (jsoref) [inline monitoring/ceph-mixin/tests_dashboards/util.py:88]: Expanding the contraction was explicitly requested https://github.com/ceph/ceph/pull/46824?notification_referrer_id=NT_kwDOACBWLLIzODY3MDU2MDc2OjIxMTkyMTI#discussion_r905779516

File: src/pybind/mgr/dashboard/controllers/host.py
Comment (jsoref) [inline src/pybind/mgr/dashboard/controllers/host.py:127]: I typically tag things like this w/ `brand`, but this isn't technically a brand. It is amusing, as #51226 is a separate PR at this point.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts
Comment (jsoref) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.spec.ts:531]: check-spelling (the tool used to identify most of these items) can apply additional rules, e.g. the one here that complained about doubled words. (It could also be used to complain about contractions if someone were so inclined.)

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/fixtures/smart_data_version_1_0_nvme_response.json
Comment (jsoref) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/fixtures/smart_data_version_1_0_nvme_response.json:82]: It's possible (likely?) this is raw captured data and shouldn't be changed.

If files come from elsewhere / shouldn't be changed, I can add them to an excludes file so that future runs should ignore them.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.ts
Comment (jsoref) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.ts:197]: this is likely to break localizations

File: src/pybind/mgr/dashboard/frontend/src/app/shared/rxjs/operators/page-visibility.operator.ts
Comment (jsoref) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/rxjs/operators/page-visibility.operator.ts:1]: notable

```diff
-visibilty
+visibility
```

see below for the consumer

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/motd-notification.service.ts
Comment (jsoref) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/services/motd-notification.service.ts:8]: see above for rename

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet
Comment (epuertat) [reply monitoring/ceph-mixin/prometheus_alerts.libsonnet:675]: Maybe we had some unit-test checking the alert message... But I'm fine with this!

File: monitoring/ceph-mixin/tests_alerts/test_alerts.yml
Comment (epuertat) [reply monitoring/ceph-mixin/tests_alerts/test_alerts.yml:1757]: I'm happy with this being fixed. IMHO CamelCase alert names (formerly we used snake_case) reduced readability and made it easier to make mistakes.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/fixtures/smart_data_version_1_0_nvme_response.json
Comment (epuertat) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/fixtures/smart_data_version_1_0_nvme_response.json:82]: Yeah, probably the source data already had this typo... We could skip this one, but either way I don't think this is going to have any impact.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.ts
Comment (epuertat) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.ts:197]: Might be, but it'd be a wrong i18n, so better to have it fixed :D

File: src/pybind/mgr/dashboard/frontend/src/app/shared/rxjs/operators/page-visibility.operator.ts
Comment (epuertat) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/rxjs/operators/page-visibility.operator.ts:1]: Does it also correct filenames? Interesting!

File: src/pybind/mgr/dashboard/plugins/lru_cache.py
Comment (epuertat) [inline src/pybind/mgr/dashboard/plugins/lru_cache.py:22]: This is intentional to comply with Python functools API:

https://github.com/python/cpython/blob/1c0a9c5a1c84bc334f2bde9d45676f19d9632823/Lib/functools.py#L430

File: src/pybind/mgr/dashboard/plugins/ttl_cache.py
Comment (epuertat) [inline src/pybind/mgr/dashboard/plugins/ttl_cache.py:27]: Same here.

**Review by epuertat — APPROVED** [review]
Everything makes sense to me here (except for the `currsize` change)! Let's ensure that all CI tests pass and this get approvals from @ceph/dashboard team!

Thank you very much @jsoref for such a rapid response!

File: src/pybind/mgr/dashboard/frontend/src/app/shared/rxjs/operators/page-visibility.operator.ts
Comment (jsoref) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/rxjs/operators/page-visibility.operator.ts:1]: The tool can complain about them: https://github.com/check-spelling/check-spelling/wiki/Feature:-Check-filenames-and-paths (it doesn't by default, although my starter workflow includes it: https://github.com/check-spelling/spell-check-this/blob/7a80f677ab5afb1e23cdaeea86e3b859abc401fe/.github/workflows/spelling.yml#L99 -- I'm probably pretty close to making it a default and letting people opt out instead of opting in)

Fixes are all by hand/with scripts.

**Comment by epuertat** [comment]
@jsoref: There are 2 failures in `make check` related to this PR and the files `monitoring/ceph-mixin/prometheus_alerts.libsonnet` and `monitoring/ceph-mixin/prometheus_alerts.yml`:
* [`Unit tests are incomplete. Tests missing for the following alerts; CephPGUnavilableBlockingIO`](https://jenkins.ceph.com/job/ceph-pull-requests/114395/testReport/junit/projectroot.monitoring/ceph-mixin/run_tox_alerts_lint/). This is failing because the changes to the alert name were removed, but the tests have been amended. Why not keeping the fixed spell for that?
* Same thing: [`Unit Testing:  /home/jenkins-build/build/workspace/ceph-pull-requests/monitoring/ceph-mixin/tests_alerts/test_alerts.yml`](https://jenkins.ceph.com/job/ceph-pull-requests/114395/testReport/junit/projectroot.monitoring/ceph-mixin/run_promtool_unittests/). `availabililty` should be corrected to `availability` in the alert, and same again for `CephPGUnavailableBlockingIO`.

The other 2 failing tests are unrelated to this.

**Comment by epuertat** [comment]
Thanks for fixing those @jsoref!

The [`make check` failure](https://jenkins.ceph.com/job/ceph-pull-requests/114419/testReport/projectroot.home.jenkins-build.build.workspace.ceph-pull-requests.src.pybind/mgr/setup_venv_for_mgr/) is unrelated, however it's blocking so until fixed, this can't be merged.

@ceph/dashboard @nizamial09: can you guys plz take care of it (although that comes from `rook` and `k8sevents` mgr modules, the [last fix to the `requirements.txt`](https://github.com/ceph/ceph/commit/f54ea99cabdaaa93a8fd8dfa6f47c807ebb33452) was yours @nizamial09)? Thanks!

```
INFO: pip is looking at multiple versions of requests to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 3) and kubernetes because these package versions have conflicting dependencies.

The conflict is caused by:
    kubernetes 11.0.0 depends on urllib3>=1.24.2
    requests 2.29.0 depends on urllib3<1.27 and >=1.21.1
```
TBH I don't understand why `pip` is unable to reconcile that... But one thing we could do is pinning deps to minor or even just major version (`kubernetes ~= 11.0`) instead of strict matching.

**Comment by jsoref** [comment]
Thanks @epuertat. The 😕 reaction is to the pip conflict which sure seems like something a solver could solve. I'm glad that I don't have to be the one to try to fix it. I'll probably look back at this on Monday or Tuesday.

**Comment by nizamial09** [comment]
thanks for all the efforts here @jsoref ! all the required checks are passing. so merging it now.

**Pull request merged**
